### ISSUE TITLE
bug: modal, popover dismiss when ion-select is dismissed

### ISSUE THREAD

**Issue by petrot** [ionic-team/ionic-framework#25124]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

I have a modal page with some inputs and drop-downs (ion-selects). When I click to the OK button after selecting an option, the click triggers the modal dismiss. 

I could catch the event on the event listener:
```
const modal = await this._modalCtrl.create({
  component: RecordEditPage,
 ...
});
 
modal.onDidDismiss().then(() => {
  console.error('dismissed'); // I see it on the console
});

await modal.present();
 ```
 
 When I set the `canDismiss` property to false, the modal works as expected, but I can't close it through `this._modalCtrl.dismiss();`

### Expected Behavior

Don't close the modal  on ion-select changes.

### Steps to Reproduce

Create a component with some ion-select input. Open the modal with the modalController.

### Ionic Info

Ionic:

   Ionic CLI                     : 6.19.0 
   Ionic Framework               : @ionic/angular 6.1.0
   @angular-devkit/build-angular : 13.3.2
   @angular-devkit/schematics    : 13.3.2
   @angular/cli                  : 13.3.2
   @ionic/angular-toolkit        : 6.1.0

Capacitor:

   Capacitor CLI      : 3.4.3
   @capacitor/android : 3.4.3
   @capacitor/core    : 3.4.3
   @capacitor/ios     : 3.4.3

Utility:

   cordova-res : 0.15.4
   native-run  : 1.5.0

System:

   NodeJS : v14.17.0  
   npm    : 8.6.0
   OS     : macOS Monterey

**Comment by liamdebeasi** [comment]
Thanks for the issue. Can you try the following dev build and let me know if it resolves the issue?

```
npm install @ionic/angular@6.1.1-dev.11649884594.1ea2f341
```

**Comment by petrot** [comment]
@liamdebeasi thanks for the quick answer! Yes, the linked dev version works perfectly!

**Comment by mattsteve** [comment]
Please make the same fix for @ionic/vue

**Comment by liamdebeasi** [comment]
The dev build applies to all of our Ionic packages. I used Angular as an example because that was what the issue author was using.

```
npm install @ionic/vue@6.1.1-dev.11649884594.1ea2f341 @ionic/vue-router@6.1.1-dev.11649884594.1ea2f341
```

**Comment by petrot** [comment]
Thank you guys, you're awesome!

**Comment by StarleyDev** [comment]
I don't like to use dev versions in production, bug in this case i'm forced to youse for fix this error... 

`npm install @ionic/angular@6.1.1-dev.11649884594.1ea2f341`

Solved the problem for now!

**Comment by liamdebeasi** [comment]
We don't recommend using dev builds in production. The dev build is to help unblock development of applications. 

If this issue is impacting a production app, we recommend rolling back the app to 6.0.16 until an update goes live.

**Comment by vicb** [comment]
Here is a repro on JS ionic https://codepen.io/vicb-the-lessful/pen/yLpRjwx

Can't wait for the fix and thanks for reacting quickly !

**Comment by aliharakeh** [comment]
I also had this issue. A quick workaroud is to add `(ionDismiss)="$event.stopPropagation()"` to `ion-select`

**Comment by liamdebeasi** [comment]
Thanks for the issue. This has been resolved via https://github.com/ionic-team/ionic-framework/pull/25125, and a fix will be available in an upcoming release of Ionic Framework.

**Comment by liamdebeasi** [comment]
A fix for this has been released in Ionic v6.1.1.
